searcher/searcher.go

```diff
@@ -211,6 +211,12 @@ func MakeAll(cfg *config.Config) (map[string]*Searcher, map[string]error, error)
 	return searchers, errs, nil
 }
 
+func PollAll(cfg *config.Config, searchers map[string]*Searcher) {
```

Comment:
This looks pretty good, but PollAll shouldn't be an exported function as there is currently no use for a Searcher that is not actively polling. Instead of exposing the notion of an inactive Searcher it would be better to have MakeAll construct all the searchers (without the background poller, like you have it) and then start the polling on each of them at the end of that function.

Something like this maybe:

```
for _, searcher := range searchers {
  go searcher.poll()
}

return searchers, errs, nil
```